### PR TITLE
user/carla: new package

### PR DESCRIPTION
**Pull request by z-erica**
note that this is headless (the gui requires qt5)

### CODE REVIEW

File: user/carla/patches/clang.patch [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+--- a/source/tests/Makefile	2024-09-21 23:56:30.000000000 +0200
++++ b/source/tests/Makefile	2024-10-21 12:19:19.555653291 +0200
+@@ -30,22 +30,20 @@
+ PEDANTIC_FLAGS  = -pedantic -pedantic-errors
+ PEDANTIC_FLAGS += -Wall -Wextra -Werror -pipe
+ 
+-PEDANTIC_FLAGS += -Wabi=98 -Wcast-qual -Wclobbered -Wconversion -Wdisabled-optimization
+-PEDANTIC_FLAGS += -Wcast-align -Wcast-qual -Wclobbered -Wconversion -Wdisabled-optimization
+-PEDANTIC_FLAGS += -Wdouble-promotion -Wfloat-equal -Wlogical-op -Wpointer-arith -Wsign-conversion
++PEDANTIC_FLAGS += -Wcast-qual -Wconversion -Wdisabled-optimization
```
Comment (nekopsykose) [inline user/carla/patches/clang.patch:29]: these don't seem to do anything and it builds either way

Comment (nekopsykose) [reply user/carla/patches/clang.patch:29]: they also build, you just have to remove Werror instead

Comment (nekopsykose) [reply user/carla/patches/clang.patch:29]: even better:

```
make_build_args = ["PEDANTIC_CFLAGS=", "PEDANTIC_CXXFLAGS="]
```

(same for make_check)

**Comment by sonata-chen** [comment]
Carla can now be compiled with Qt6, although it's not officially released yet.
https://github.com/sonata-chen/cports/commit/a1cfb210c179003f19819b5c654f89d989e5cf43

**Pull request merged**